File: 1711-find-valid-matrix-given-row-and-column-sums/find-valid-matrix-given-row-and-column-sums.java
class Solution {
    public int[][] restoreMatrix(int[] row, int[] col) {
        int[][] res = new int[row.length][col.length];
        int m;
        for(int i = 0;i <row.length;i++){
            for(int j = 0;j<col.length;j++){
                m = Math.min(row[i],col[j]);
                res[i][j] = m;
                row[i] -=m;
                col[j] -=m;  
            }
        }   
        return res;
    }
}

File: 1711-find-valid-matrix-given-row-and-column-sums/find-valid-matrix-given-row-and-column-sums.py
class Solution:
    def restoreMatrix(self, row: List[int], col: List[int]) -> List[List[int]]:
        
        res =[]
        for i in range(len(row)):
            res.append([])
            for j in range(len(col)):
                m = min(row[i],col[j])
                res[-1].append(m)
                row[i]-=m
                col[j]-=m
        return res